src/opening-period/page/EditOpeningPeriodPage.test.tsx

```diff
@@ -0,0 +1,332 @@
+          modified: '2020-12-18T10:00:00.000000+02:00',
+          name: {
+            fi: null,
+            sv: null,
+            en: null,
+          },
+          resource_state: ResourceState.SELF_SERVICE,
+          start_time: '12:00:00',
+          weekdays: [6, 7],
+        },
+        {
+          created: '2020-12-18T10:00:00.000000+02:00',
+          description: {
+            fi: 'Arkena ovet menevät kiinni puolta tuntia aiemmin',
+            sv: null,
+            en: null,
+          },
+          end_time: '18:00:00',
+          full_day: false,
+          group: 1225,
+          id: 2636,
+          modified: '2020-12-18T10:00:00.000000+02:00',
+          name: {
+            fi: null,
+            sv: null,
+            en: null,
+          },
+          resource_state: ResourceState.OPEN,
+          start_time: '10:00:00',
+          weekdays: [1, 2, 3, 4, 5],
+        },
+      ],
+    },
+  ],
+};
+
+describe(`<EditNewOpeningPeriodPage />`, () => {
+  beforeEach(() => {
+    jest
+      .spyOn(api, 'getResource')
+      .mockImplementation(() => Promise.resolve(testResource));
+
+    jest
+      .spyOn(api, 'getDatePeriodFormOptions')
+      .mockImplementation(() => Promise.resolve(testDatePeriodOptions));
+
+    jest
+      .spyOn(api, 'getDatePeriod')
+      .mockImplementation(() => Promise.resolve(testDatePeriod));
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should render basic date-period data', async () => {
+    let container: HTMLElement;
+
+    await act(async () => {
+      container = render(
+        <EditOpeningPeriodPage resourceId="tprek:8100" datePeriodId="1" />
+      ).container;
+
+      if (!container) {
+        throw new Error(
+          'Something went wrong in rendering of CreateNewOpeningPeriodPage'
+        );
+      }
+    });
+
+    await act(async () => {
+      // Check info data
+      expect(
+        container.querySelector('[data-test="openingPeriodTitle"]')
+      ).toHaveValue(testDatePeriod.name.fi);
+
+      expect(
+        container.querySelector('#openingPeriodOptionalDescription')
+      ).toHaveValue(testDatePeriod.description.fi);
+
+      expect(container.querySelector('#openingPeriodBeginDate')).toHaveValue(
+        '18.12.2020'
+      );
+      expect(container.querySelector('#openingPeriodEndDate')).toHaveValue(
+        '18.01.2021'
+      );
+    });
+  });
+
+  it('should render first the time-span when it has the weekdays in the beginning of the week', async () => {
+    let container: HTMLElement;
+
+    await act(async () => {
+      container = render(
+        <EditOpeningPeriodPage resourceId="tprek:8100" datePeriodId="1" />
+      ).container;
+
+      if (!container) {
+        throw new Error(
+          'Something went wrong in rendering of CreateNewOpeningPeriodPage'
+        );
+      }
+    });
+
+    await act(async () => {
+      expect(
+        container.querySelector('[data-test="weekdays-monday-0-checkbox"]')
```

Comment:
How do we know that this element is first? Is it because of the index value 0? Im wondering if there could be some other way to verify, that those elements are before other elements in dom?

Reply:
I changed the list to use the <ul> and <li> elements. Now the first or last <li> is used in assertions as a root element.